tokio/src/runtime/metrics/runtime.rs

```diff
@@ -0,0 +1,448 @@
+use crate::runtime::Handle;
+
+use std::sync::atomic::Ordering::Relaxed;
+use std::time::Duration;
+
+/// Handle to the runtime's metrics.
+///
+/// This handle is internally reference-counted and can be freely cloned. A
+/// `RuntimeMetrics` handle is obtained using the [`Runtime::metrics`] method.
+///
+/// [`Runtime::metrics`]: crate::runtime::Runtime::metrics()
+#[derive(Clone, Debug)]
+pub struct RuntimeMetrics {
```

Comment:
I wonder if it makes more sense to have a `Worker` struct that contains `Handle` and `worker_id: usize` instead of having each api panic if the index is incorrect?

Reply:
Tried that initially. It is a pain to wire it all up.